webapp-django/crashstats/crashstats/tests/test_views.py

```diff
@@ -407,42 +407,74 @@ def mocked_get(url, params, **options):
                  }
                       """ % {'end_date': now.strftime('%Y-%m-%d'),
                              'yesterday': yesterday.strftime('%Y-%m-%d')})
-            if '/supersearch/fields/' in url:
-                results = copy.copy(SUPERSEARCH_FIELDS_MOCKED_RESULTS)
-                # to be realistic we want to introduce some dupes
-                # that have a different key but its `in_database_name`
-                # is one that is already in the hardcoded list (the
-                # baseline)
-                assert 'accessibility' not in results
-                results['accessibility'] = {
-                    'name': 'accessibility',
-                    'query_type': 'string',
-                    'namespace': 'raw_crash',
-                    'form_field_choices': None,
-                    'permissions_needed': [],
-                    'default_value': None,
-                    'is_exposed': True,
-                    'is_returned': True,
-                    'is_mandatory': False,
-                    'in_database_name': 'Accessibility',
-                }
-                return Response(results)
             raise NotImplementedError(url)
 
         rget.side_effect = mocked_get
 
+        def mocked_supersearchfields(**params):
+            results = copy.copy(SUPERSEARCH_FIELDS_MOCKED_RESULTS)
+            # to be realistic we want to introduce some dupes
+            # that have a different key but its `in_database_name`
+            # is one that is already in the hardcoded list (the
+            # baseline)
+            assert 'accessibility' not in results
+            results['accessibility'] = {
+                'name': 'accessibility',
+                'query_type': 'string',
+                'namespace': 'raw_crash',
+                'form_field_choices': None,
+                'permissions_needed': [],
+                'default_value': None,
+                'is_exposed': True,
+                'is_returned': True,
+                'is_mandatory': False,
+                'in_database_name': 'Accessibility',
+            }
+            return results
+
+        self.supersearch = mock.MagicMock()
+        self.supersearchfields = mock.MagicMock()
+        self.missingfields = mock.MagicMock()
+
+        from crashstats.supersearch.models import (
+            SuperSearch,
+            SuperSearchFields,
+            SuperSearchField,
+            SuperSearchMissingFields,
+        )
+        # monkey-patch a class attribute
+        SuperSearchFields.implementation = self.supersearchfields
+        SuperSearchField.implementation = self.supersearchfields
+        SuperSearchMissingFields.implementation = self.missingfields
+        SuperSearch.implementation = self.supersearch
```

Comment:
This is tricky. It really is. 
So, it started with I needed to patch (aka. monkey-patch) the supersearchfields because it's needed by every test almost since it depends on it for whitelisting. Then, calling `SuperSearchFields().get()` at least once makes sure the internal cache is pre-populated. 

However, the patching of `SuperSearchField` (note lack of s suffix), `SuperSearchMissingFields` and `SuperSearch` seems to be widely used. 

I commented these out to see if I can move them into places where they belong (e.g. `crashstats/supersearch/tests/`) but when I did that (just the commenting out part) the following files fail:
```
crashstats/api/tests/test_views.py
crashstats/crashstats/tests/test_views.py
crashstats/manage/tests/test_views.py
crashstats/signature/tests/test_views.py
supersearch/tests/test_views.py
```
22 tests stopped working. Spread across **5 apps**. I guess it goes to show the this is now all over the whole web app project :)

But I also agree with what you say. It is a weird place to do it. So what I'm going to do is to move this functionality into `DjangoTestCase` and not do it in the `setUp` but something like `def get_mocked_supersearch(self)` so it's a bit more explicit.